src/ode_solver/pod_galerkin_ode_solver.h

```diff
@@ -16,14 +16,14 @@ template <int dim, typename real, typename MeshType = dealii::Triangulation<dim>
 #else
 template <int dim, typename real, typename MeshType = dealii::parallel::distributed::Triangulation<dim>>
 #endif
-class PODGalerkinODESolver: public ODESolverBase <dim, real, MeshType>
+class PODGalerkinODESolver: public ImplicitODESolver<dim, real, MeshType>
```

Comment:
This is a pretty tricky one. Looking at it from the point of view of has-a vs is-a, I would argue that the Galerkin and Petrov-Galerkin ODE solvers are an implicit ODE solver. They are fundamentally using the same implicit scheme to solve the equation, but with a matrix multiplication to project to a different subspace. Once they are created using a POD basis, they can be used interchangeably with an implicit solver anywhere in the code without any issue which is the main idea behind the LSP. The only thing is that the solution obtained would be different, or if a bad POD basis is used, the solution might not converge. Therefore, in that sense, the preconditions are different and the reduced-order implicit solvers require a POD basis that properly describes the solution space. So the LSP might not be strictly respected. 

However, the constructors for the reduced-order implicit solvers require a POD basis. Therefore, it would have to be assumed that if a reduced-order implicit solver is being created and used in the place of an implicit solver, the user is fully aware of the preconditions. Otherwise, the reduced-order implicit solver wouldn't be getting created in the first place. The same goes for the factory class, it's not possible to create a reduced-order solver unless you're giving it a POD basis.

Owning an ImplicitODESolver brings a series of new problems because of the current inheritance-based design. In the reduced-order solver wouldn't be able to interface with the private or protected members of the implicit solver in its current form. It would have been possible to make a composition-based solution, but that would likely involve re-writing the entire ODE solver section.